gpu/octree/src/utils/approx_nearest_utils.hpp

```diff
@@ -0,0 +1,127 @@
+/*
+ * SPDX-License-Identifier: BSD-3-Clause
+ *
+ *  Point Cloud Library (PCL) - www.pointclouds.org
+ *  Copyright (c) 2020-, Open Perception
+ *
+ *  All rights reserved
+ */
+
+#pragma once
+
+#include "morton.hpp"
+#include <assert.h>
+
+#include <limits>
+#include <tuple>
+#include <bitset>
+
+namespace pcl {
+namespace device {
+
+__device__ __host__ __forceinline__ int
+getBitsNum(const int integer)
+{
+  return std::bitset<8*sizeof(integer)> (integer).count();
+}
```

Comment:
`std::bitset` cannot be used inside a `__device__` function
```suggestion
__device__ __host__ __forceinline__ unsigned
getBitsNum(const unsigned integer)
{
  #ifdef __CUDA_ARCH__
    return __popc(integer);
  #else
    return std::bitset<8*sizeof(integer)> (integer).count();  
  #endif
}